src/server/routes/consentToken.ts

```diff
@@ -0,0 +1,73 @@
+import { Request } from 'express';
+import { ResponseWithRequestState } from '@/server/models/Express';
+import { renderer } from '@/server/lib/renderer';
+import { rateLimitedTypedRouter as router } from '@/server/lib/typedRoutes';
+import { handleAsyncErrors } from '@/server/lib/expressWrappers';
+import {
+  validateConsentToken,
+  resendConsentEmail,
+} from '@/server/lib/idapi/consentToken';
+import { mergeRequestState } from '@/server/lib/requestState';
+
+// This route is of this specific form because it's a direct copy of the legacy
+// route in identity-frontend, and emails sent by IDAPI contain this URL.
+// We can change the route after we've migrated the emails from IDAPI.
+router.get(
+  '/consent-token/:token/accept',
+  handleAsyncErrors(async (req: Request, res: ResponseWithRequestState) => {
+    const ip = req.ip;
+    const sc_gu_u = req.cookies.SC_GU_U;
+    const token = req.params.token;
+    try {
+      await validateConsentToken(ip, sc_gu_u, token, res.locals.requestId);
+      // Redirect to /consents/thank-you (a page managed by Frontend). This is
+      // to retain the legacy behaviour of the route from identity-frontend.
+      return res.redirect('/consents/thank-you');
```

Comment:
Slightly pedantic, but we should use 303 redirects as that makes the most sense: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/303

Rather than the Express default which is 301: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301

Same suggestion for the other redirects in this file.

```suggestion
      return res.redirect(303, '/consents/thank-you');
```